community/howto/EricZimmerman/BackupConfigToGithub.md

```diff
@@ -0,0 +1,304 @@
+
+    NOTE: If you are prompted to reauthenticate, do so. This is a safety measure as you are about to create a sensitive credential
+
+    ![](img/newToken.png)
+
+3. Give your token a meaningful name, set the **Expiration** date to `No expiration`, and check the `workflow` and `read:org` under `admin:org` section
+
+    :warning: If you prefer to set an expiration date for your token, feel free to do so, but be aware you will have to regenerate a new token as things expire, as well as update some other settings that we have not got to yet :warning:
+
+    ![](img/tokenSettings.png)
+
+4. Scroll to the bottom of the page, and click `Generate token`
+
+    ![](img/generateToken.png)
+
+5. You now have a new access key! Protect it well, like you would any other password! Click on the `Copy` button to copy the token to the clipboard, then paste it into the same place you saved your HTTPS URL from the previous step. 
+   
+    This is your **ONE AND ONLY** chance to copy this key, so make sure you record it before leaving this screen.
+
+    NOTE: The image below has the key redacted. Yours will of course not look like that
+
+    ![](img/copyToken.png)
+
+### Combine our HTTPS URL with our token
+
+We can now put the pieces together we have created so far. To make this easier, I will be working with these pieces, but you should of course use the values you generated above.
+
+- **URL**: https://github.com/EricZimmerman/Voron24Configs.git
+- **Token**: ghp_dKXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+1. Insert the token into the URL following this pattern:  
+
+    ```
+    https://<Token>@github.com/EricZimmerman/Voron24Configs.git
+    ```
+
+    which for my values, would end up looking like:
+
+    ```
+    https://ghp_dKXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@github.com/EricZimmerman/Voron24Configs.git
+    ```
+
+    Note that you do not want the <> symbols, as that's just a placeholder so you can see the pattern easier.
+
+2. Copy this newly crafted URL to your text editor. This is the URL we will be using in a later step.
+
+    :warning: :warning: :warning: **NEVER EVER** put the token you generated in any of the files that will be committed to GitHub! If you do so, GitHub will see it and (rightly) revoke your token! :warning: :warning: :warning:
+
+
+## Connect to your printer via SSH
+
+1. Open up a PowerShell window, or use whatever program you usually use (many people use **putty**, but ssh is built right into Windows) to connect to your printer via SSH, and connect to your printer
+
+    ![](img/InitialSsh.png)
+
+2. Change directories into your `config` directory, which is where all of your configuration files live
+
+    ```bash
+    cd ~/printer_data/config
+    ```
+    
+3. Using nano, or your favorite command line based text editor (its nano, right?), create new file named `autocommit.sh` in your `config` directory
```

Comment:
Is this step necessary? The next step will download the file directly in the config directory.

Reply:
ah yes this is a left over from when i didnt have them download it directly, but create it. will clean that up too